app/controllers/users_controller.rb

```diff
@@ -0,0 +1,58 @@
+class UsersController < ApplicationController
+
+  get'/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+
+    erb :'users/show_tweet'
+  end
+
+  # -- signup
+  get "/signup" do
+    if !session[:user_id]
+      erb :"users/create_user"
+
+    else
+      redirect to '/tweets'
+    end
+  end
+
+  post "/signup" do
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      redirect to "/signup"
+    else
+      @user = User.new( username: params[:username], email: params[:email], password: params[:password])
```

Comment:
just use the `create` method here so you don't need two lines of code to do the same thing.